pkg/apis/v1alpha1/ipaddressallocation_types.go

```diff
@@ -0,0 +1,62 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	VisibilityExternal = "External"
+	VisibilityPrivate  = "Private"
+	VisibilityProject  = "Project"
+)
+
+// +genclient
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:storageversion
+
+// IPAddressAllocation is the Schema for the ipallocation API.
+// +kubebuilder:printcolumn:name="IPAddressBlockVisibility",type=string,JSONPath=`.spec.ip_address_block_visibility`,description="IPAddressBlockVisibility of IPAddressAllocation"
+// +kubebuilder:printcolumn:name="CIDR",type=string,JSONPath=`.status..cidr`,description="CIDRs for the IPAddressAllocation"
+type IPAddressAllocation struct {
```

Comment:
use IPAddressAllocations?

Reply:
@dantingl @jianjuns What's your opinion?

Reply:
NSX API use IPAddressAllocation as request resource type, so I prefer keeping it aligned.